### ISSUE TITLE
.getLocation(), .getGeometry(), .getFullscreen() not a thing!

### ISSUE THREAD

**Issue by FlaggedBrad** [jarvisteach/appJar#236]
I would love to be able to get the state of a window. sadly, this feature is nonexistent

**Comment by jarvisteach** [comment]
You're right - nothing like that exists. There's never been much of a use case... I'd love to know what you're trying to achieve.

You can access tkinter direct though, to get information.

This will give you the location & dimensions: `app.topLevel.geometry()`
And this will tell you if you're in full screen: `app.toplevel.attribute('-fullscreen')`

**Comment by FlaggedBrad** [comment]
I'm trying to save the size of the window, so that when the next time the user opens the window, I can restore the window size of the last session.

**Comment by jarvisteach** [comment]
Hmm, that's a nice feature to build in.

appJar could create a hidden file that stores settings like that, and update it each time `.stop()` is called.

This could be turned off by default, but the user could call a function to enable it, passing an optional file name (otherwise it could be `.appjar`)

This could contain:
* height & width
* top left X & Y
* fullscreen True/False
* toolbar toggle
* various container settings: tab/page number, paned/scrolled window positions, toggle frame settings

appJar could also then provide an API to get/set user definable properties...

What else might be useful?

**Comment by FlaggedBrad** [comment]
Oh my God, this would be absolutely perfect. I can't really think of anything else that would make it better.

**Comment by jarvisteach** [comment]
As mentioned in #265 - settings should be stored in a platform specific config location.

**Comment by jarvisteach** [comment]
Basic concept working. Now need to build it in.
* Command line parameter - to load a config file
* Constructor call to load and (auto) save config file
* stop function will need to check if is saving
* function to set config file name/location

Need to standardise code:
* refactor language config file and settings config file into one thing, and merge functions
* Make the loaded config file a global variable, the user can then set fields, and when they call save - choose to update or keep originals

Generic fields:
* Add functions to get/set generic fields

**Comment by jarvisteach** [comment]
When settings are saved - if a new window size has been set, it will be loaded and set as the minimum - we need to save the minimum and use that instead...
